File: examples opp/7_function_overiding.py
class animal:
    def live(self):
        print("i live in jungle")

    def sound(self):
        pass


class cat(animal):
    def sound(self):
        print("mew mew")


class monkey(animal):
    def sound(self):
        print("koyak koyak")


monkey1 = monkey()
monkey1.sound()
monkey1.live()
cat1 = cat()
cat1.sound()

File: examples opp/7_function_overiding.cpp
#include <bits/stdc++.h>
using namespace std;
class animal
{
public:
    void sound()
    {
        cout << "sound nai" << endl;
    }
};
class cat : public animal
{
public:
    void sound()
    {
        cout << "mew mew" << endl;
    }
};
class dog : public animal
{
public:
    void sound()
    {
        cout << "gau gau" << endl;
    }
};

int main()
{
    dog german;
    german.sound();
    cat ikri;
    ikri.sound();
    return 0;
}